frontend/src/scenes/session-recordings/playlist/sessionRecordingsPlaylistLogic.ts

```diff
@@ -788,19 +798,46 @@ export const sessionRecordingsPlaylistLogic = kea<sessionRecordingsPlaylistLogic
             if (values.sessionRecordingsResponseLoading) {
                 return // We don't want to load if we are currently loading
             }
+
             actions.loadSessionRecordings(direction)
         },
 
         loadSessionRecordingsSuccess: () => {
             actions.maybeLoadPropertiesForSessions(values.sessionRecordings)
         },
 
-        setSelectedRecordingId: () => {
+        setSelectedRecordingId: async () => {
             // Close filters when selecting a recording
             actions.setIsFiltersExpanded(false)
 
-            // If we are at the end of the list then try to load more
             const recordingIndex = values.sessionRecordings.findIndex((s) => s.id === values.selectedRecordingId)
+
+            // If recording not found in current list, fetch it specifically
+            if (recordingIndex === -1 && values.selectedRecordingId) {
+                try {
+                    // Fetch the specific recording using session_ids filter
+                    const response = await api.recordings.list({
+                        ...convertUniversalFiltersToRecordingsQuery(values.filters),
+                        session_ids: [values.selectedRecordingId],
+                    })
+
+                    if (response.results && response.results.length > 0) {
+                        // Prepend the recording to the list so it's visible
+                        actions.prependRecording(response.results[0])
+                    } else {
+                        lemonToast.error('Recording not found')
+                        actions.setSelectedRecordingId(null)
```

Comment:
this (effective) navigation threw me when i tried it locally since it changed the URL, i wonder if it's better to not clear here but have some sensible UI state so someone can still copy the URL and ask "hey wwhy didn't this work?"